lib/asciidoctor/iso/cleanup.rb

```diff
@@ -86,6 +88,42 @@ def sort_biblio(bib)
         end
       end
 
+      def termdef_cleanup(xmldoc)
+        generate_termref_links(xmldoc)
+        super
+      end
+
+      def generate_termref_links(xmldoc)
+        termlookup = replace_automatic_genrated_ids_terms(xmldoc)
+        xmldoc.xpath('//termxref').each do |node|
+          target = node.text
+          next if termlookup[target].nil?
+
+          node.name = 'xref'
+          node['target'] = termlookup[target]
+        end
+      end
+
+      def replace_automatic_genrated_ids_terms(xmldoc)
+        xmldoc.xpath('//term').each.with_object({}) do |term_node, termlookup|
+          next if AUTOMATIC_GENERATED_ID_REGEXP.match(term_node['id']).nil?
+
+          term_text = term_node.at('./preferred').text
+          term_node['id'] = unique_text_id(xmldoc, term_text)
+          termlookup[term_text] = term_node['id']
+        end
+      end
+
+      def unique_text_id(xmldoc, text)
+        return "text-#{text}" if xmldoc.at("//*[@id = 'text-#{text}']").nil?
```

Comment:
"text-#{text}" should be "term-#{text}" throughout this method

Reply:
Done